.github/workflows/release-upgradeable.yml

```
@@ -0,0 +1,103 @@
+name: Release Upgradeable
+
+on:
+  workflow_dispatch: {}
+
+jobs:
+  state:
+    name: Check state
+    permissions:
+      pull-requests: read
+    if: ${{ !endsWith(github.repository, '-upgradeable') }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v6
+        with:
+          repository: ${{ github.repository }}-upgradeable
+          ref: ${{ github.ref }}
+      - uses: actions/checkout@v6
+        with:
+          ref: ${{ github.ref }}
+          path: lib/openzeppelin-contracts
+      - name: Set up environment
+        uses: ./.github/actions/setup
+      - name: Get reference commit
+        id: get-reference-commit
+        run: |
+          cd lib/openzeppelin-contracts
+          echo "commit=$(git rev-parse --short HEAD)" >> "$GITHUB_OUTPUT"
+      - name: Check upgradeable
+        id: check-upgradeable
+        run: bash scripts/release/workflow/check-upgradeable.sh
+        env:
+          REFERENCE_COMMIT: ${{ steps.get-reference-commit.outputs.commit }}
+    outputs:
+      publish: ${{ steps.check-upgradeable.outcome }}
+      is_prerelease: ${{ steps.check-upgradeable.outputs.is_prerelease }}
+
+  # copied from release-cycle.yml
+  publish:
+    needs: state
+    name: Publish to npm
+    environment: push-upgradeable
+    permissions:
+      contents: write
+      id-token: write
+    if: needs.state.outputs.publish == 'success' # Note: changed from 'true' to 'success' to support the way publish is computed
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v6
+        with:
+          repository: ${{ github.repository }}-upgradeable
+          ref: ${{ github.ref }}
+      - uses: actions/checkout@v6
+        with:
+          ref: ${{ github.ref }}
+          path: lib/openzeppelin-contracts
+      - name: Set up environment
+        uses: ./.github/actions/setup
+      - id: pack
+        name: Pack
+        run: bash scripts/release/workflow/pack.sh
+        env:
+          PRERELEASE: ${{ needs.state.outputs.is_prerelease }}
+      - name: Upload tarball artifact
+        uses: actions/upload-artifact@v5
+        with:
+          name: ${{ github.ref_name }}
```

Maybe we should prefix the suffix:

```suggestion
          name: ${{ github.ref_name }}-upgradeable
```

This would avoid overwriting the vanilla tarball artifact. Not a big deal but I wonder if it may become an issue if (let's say) we run publishing for vanilla and upgradeable in parallel